src/xo-app/home/index.js

```diff
@@ -232,13 +243,29 @@ export default class Home extends Component {
   }
 
   componentWillReceiveProps (props) {
-    this._initFilterAndSortBy(props)
+    if (this._getFilter() !== this._getFilter(props)) {
+      this._initFilterAndSortBy(props)
+    }
     if (props.type !== this.props.type) {
       this.setState({ activePage: undefined, highlighted: undefined })
     }
   }
 
+  componentDidUpdate () {
+    const { selectedItems } = this.state
+
+    // Unselect items that are no longer visible
+    const newSelectedItems = pick(selectedItems, map(this._getVisibleItems(), 'id'))
+    if (size(newSelectedItems) < this._getNumberOfSelectedItems()) {
```

Comment:
Would not it be possible to improve this?

- check that visible items have really changed (`this.getVisibleItems.recomputations()` before and after calling the selector)